### ISSUE TITLE
The Open User Data Directory button in the Credits menu does nothing when clicked

### ISSUE THREAD

**Issue by Calinou** [flathub/net.minetest.Minetest#29]
**OS:** Fedora 33
**Minetest version:** 5.4.0 (Flatpak https://github.com/flathub/net.minetest.Minetest/commit/3f4149349a35f49372a5da2ce1bb54a9ee456e63)

The **Open User Data Directory** button in the Credits menu does nothing when clicked. In a non-Flatpak installation, it will open the system file manager. This is presumably caused by Flatpak sandboxing limitations, but I don't know how to fix this one.

![image](https://user-images.githubusercontent.com/180032/109062513-5413b500-76e8-11eb-8c02-e621bc4868ba.png)

**Comment by Docmine17** [comment]
same problem here.

**Comment by rubenwardy** [comment]
Looks like this is the relevant issue: https://github.com/flatpak/flatpak-xdg-utils/issues/5

**Comment by rubenwardy** [comment]
Ok, spent some time working this out. Thanks to the very friendly people in the #flatpak matrix channel, especially Owen Taylor

The problem is that the flatpak internally uses ~/.minetest for the user directory but on the host it is ~/.var/app/net.minetest.Minetest/.minetest.  When Minetest tries to open ~/.minetest, xdg-desktop-portal is unable to convert this to the host path location (the issue for it to do that is here https://github.com/flatpak/xdg-desktop-portal/issues/591 )

To fix this:

- Remove `--persist=.minetest`. This is what makes the flatpak use ~/.minetest to save data. https://github.com/flathub/net.minetest.Minetest/blob/master/net.minetest.Minetest.yaml#L15
- Change an environment variable so that Minetest uses the correct user data directory. Currently, Minetest uses HOME. In 5.7.0-dev, it supports MINETEST_USER_PATH. It's probably not a good idea to change the HOME env var, so this fix may need to wait for 5.7.0

A temporary solution to fix this is the following

```bash
flatpak override --user --env=HOME=/home/ruben/.var/app/net.minetest.Minetest net.minetest.Minetest
```
